src/scanpy/_utils/__init__.py

```diff
@@ -713,6 +713,27 @@ def _(
     )
 
 
+@singledispatch
+def axis_nnz(X: np.ndarray, axis: Literal[0, 1]) -> np.ndarray:
+    return np.count_nonzero(X, axis=axis)
+
+
+@axis_nnz.register(sparse.spmatrix)
+def _(X: sparse.spmatrix, axis: Literal[0, 1]) -> np.ndarray:
+    return X.getnnz(axis=axis)
+
+
+@axis_nnz.register(DaskArray)
+def _(X: DaskArray, axis: Literal[0, 1]) -> DaskArray:
+    return X.map_blocks(
```

Comment:
strange, codecov says this isn’t covered.

I assume it’s just lying, but please check if that message has vanished before you merge.

Reply:
<img width="514" alt="Screenshot 2024-10-30 at 09 49 45" src="https://github.com/user-attachments/assets/b4a1017b-3575-439b-b7f2-fac2b567dcb3">
gone!